client.js

```diff
@@ -46,14 +46,15 @@ function checkMessages() {
             setTimeout(checkMessages, 100)
             return
         }
-        var estimatedTime = (request.response.byteLength / 4) / 4.41 + 2
-        setTimeout(checkMessages, estimatedTime)
+//        var estimatedTime = (request.response.byteLength / 4) / 4.41 + 2
+//        setTimeout(checkMessages, estimatedTime)
 
         console.log('received ' + request.response.byteLength)
         var source = playbackCtx.createBufferSource()
         console.log('start playback')
         source.onended = function() {
             console.log('end playback')
+            setTimeout(checkMessages, 1000) // aici ar trebui setat TimeOut-ul corect
```

Comment:
Ideea e ca schimbi momentul in care repornesti functia de check. Inainte dadeai timeout la functia de verificare de mesaje din momentul in care se incarca sursa, pe cand in a doua varianta te asiguri ca a redat tot ce era de ascultat inaintea trimiterii unui nou check. E destul de dodgy, dar functioneaza.